src/libs/HttpUtils.js

```diff
@@ -64,7 +69,13 @@ function download(relativePath) {
     return processHTTPRequest(`${siteRoot}${strippedRelativePath}`);
 }
 
+function abortPendingRequests() {
+    abortController.abort();
+    abortController = new AbortController();
```

Comment:
Why is a new controller created here? The same controller can't be reused?

Reply:
I don't think we should reuse the controller or we can't make further requests. 

A comment explaining this with some reference would be nice ig.

Reply:
I know from somewhere that it's not reusable after you call `abort` but was only able to find this: https://javascript.tutorialink.com/cant-make-new-fetch-request-after-aborting-previous/

Once you call `abort`, if you try to start new requests and hook them to this already aborted controller, they just don't start

Reply:
Hmm actually found something: https://dom.spec.whatwg.org/#abortcontroller-api-integration
> Reject immediately if the [AbortSignal](https://dom.spec.whatwg.org/#abortsignal) is already [aborted](https://dom.spec.whatwg.org/#abortsignal-aborted),

Reply:
Nice find!